common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,172 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
 
-import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import android.os.Build;
+import android.support.annotation.NonNull;
 
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.PkceChallenge;
+
+import java.io.UnsupportedEncodingException;
 import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
 
-public class AzureActiveDirectoryAuthorizationRequest extends AuthorizationRequest {
+public class AzureActiveDirectoryAuthorizationRequest extends MicrosoftAuthorizationRequest {
+    private static final String TAG = AzureActiveDirectoryAuthorizationRequest.class.getSimpleName();
 
-    private URL mAuthority;
+    /* Constants */
+    private static final String RESOURCE = "resource";
+    private static final String CLIENT_REQUEST_ID = "client-request-id";
+    private static final String SCOPE_OPENID_VALUE = "openid";
+    private static final String PLATFORM_VALUE = "ADAL.Android";
+    private static final String QUERY_PROMPT_REFRESH_SESSION_VALUE = "refresh_session";
 
     /**
-     * @return URL authority
+     * The App ID URI of the target web API.
+     * This is required in one of either the authorization or token requests.
+     * To ensure fewer authentication prompts place it in the authorization request to
+     * ensure consent is received from the user.
      */
-    public URL getAuthority() {
-        return mAuthority;
-    }
-
+    private String mResource;
+    //TODO The microsoft doc is different with V1 has currently.
     /**
-     * @param authority URL
+     * Optional. Indicate the type of user interaction that is required.
      */
-    public void setAuthority(URL authority) {
-        mAuthority = authority;
+    private AzureActiveDirectoryPromptBehavior mPromptBehavior;
+    private String mClaimsChallenge;
+
+    public AzureActiveDirectoryAuthorizationRequest(final String responseType,
```

Comment:
Any thoughts about using a builder pattern to create these objects?

Reply:
I filed the refactor here. https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/142 I want to merge this PR first to unblock Krishna.